table/block_based/block_based_table_reader.cc

```diff
@@ -1935,19 +1939,36 @@ CachableEntry<FilterBlockReader> BlockBasedTable::GetFilter(
       } else {
         RecordTick(rep_->ioptions.statistics, BLOCK_CACHE_ADD_FAILURES);
         delete filter;
-        return CachableEntry<FilterBlockReader>();
+        return_empty_reader = true;
       }
     }
   }
 
+  if (block_cache_tracer_ && lookup_context) {
```

Comment:
Do we need to check the value `s`?

Reply:
As discussed offline, it may be better to log the user's intention of insertion instead of setting no_insert based on the insertion result.